library/classes/Document.class.php

```diff
@@ -490,16 +503,36 @@ function change_patient($new_patient_id) {
    * @param  string  $filename     Desired filename, may be modified for uniqueness
    * @param  string  $mimetype     MIME type
    * @param  string  &$data        The actual data to store (not encoded)
+   * @param  string  $tmpfile      The tmp location of file
    * @param  string  $higher_level_path Optional subdirectory within the local document repository
    * @param  string  $path_depth   Number of directory levels in $higher_level_path, if specified
    * @param  integer $owner        Owner/user/service that is requesting this action
    * @return string                Empty string if success, otherwise error message text
    */
-  function createDocument($patient_id, $category_id, $filename, $mimetype, &$data,
+  function createDocument($patient_id, $category_id, $filename, $mimetype, &$data,$tmpfile,
     $higher_level_path='', $path_depth=1, $owner=0) {
     // The original code used the encounter ID but never set it to anything.
```

Comment:
Two things here:
1. Can't use mimetype rather than bringing in $tmpfile(I am just looking for reasoning why doing this; if there is a reason, then would be ok).
2. Placing a new parameter in the middle seems very dangerous. Since this function is used in other places in the codebase, wouldn't it be much safer to add the parameter at the end?